python/looker_sdk/client.py

```diff
@@ -23,25 +23,102 @@
 """Client entry point
 """
 from typing import Optional
+import warnings
 
 from looker_sdk.rtl import api_settings
 from looker_sdk.rtl import requests_transport
 from looker_sdk.rtl import serialize
 from looker_sdk.rtl import auth_session
-from looker_sdk.sdk import methods
+from looker_sdk.sdk.api31 import methods as methods31
+from looker_sdk.sdk.api40 import methods as methods40
+
+
+SPECIFIC_FACTORY = (
+    "Please use a version specific function such as "
+    "client.init31() or client.init40()"
+)
+API_SETTINGS_API_VERSION_DEPRECATED = (
+    f"API_VERSION config value is no longer needed. {SPECIFIC_FACTORY}"
+)
+
+
+class SetupError(Exception):
+    pass
 
 
 def setup(
     config_file: str = "looker.ini", section: Optional[str] = None
-) -> methods.LookerSDK:
+) -> methods31.LookerSDK:
     """Default dependency configuration
     """
+    warnings.warn(
+        message=DeprecationWarning(
+            f"client.setup() will be removed in future versions, {SPECIFIC_FACTORY}"
+        )
+    )
     settings = api_settings.ApiSettings.configure(config_file, section)
+    if settings.api_version:
+        warnings.warn(message=DeprecationWarning(API_SETTINGS_API_VERSION_DEPRECATED))
+    if settings.api_version and settings.api_version != "3.1":
+        raise SetupError(
+            "You are attempting to use the Looker31SDK with an api_version config setting of '3.1'"
+        )
+    settings.api_version = "3.1"
     settings.headers = {"Content-Type": "application/json"}
+    if not settings.is_configured():
+        raise SetupError(
+            f"Missing required configuration values like base_url and api_version."
+        )
     transport = requests_transport.RequestsTransport.configure(settings)
-    return methods.LookerSDK(
+    return methods31.Looker31SDK(
+        auth_session.AuthSession(settings, transport, serialize.deserialize31),
+        serialize.deserialize31,
+        serialize.serialize,
+        transport,
+    )
+
+
+def init31(
```

Comment:
we actually need to delete this entire file and re-add the `init{31,40}` methods into `python/looker_sdk/__init__.py` I can jump on that now

Reply:
Thanks. I realize what you asked me about yesterday now!